contracts/src/TokenHome/ERC20TokenHome.sol

```diff
@@ -27,8 +27,25 @@ import {CallUtils} from "../utils/CallUtils.sol";
 contract ERC20TokenHome is IERC20TokenHome, TokenHome {
     using SafeERC20 for IERC20;
 
-    /// @notice The ERC20 token this home contract transfers to TokenRemote instances.
-    IERC20 public immutable token;
+    // solhint-disable private-vars-leading-underscore
+    /// @custom:storage-location erc7201:avalanche-ictt.storage.ERC20TokenHome
+    struct ERC20TokenHomeStorage {
+        /// @notice The ERC20 token this home contract transfers to TokenRemote instances.
+        IERC20 _token;
+    }
+    // solhint-enable private-vars-leading-underscore
+
+    // keccak256(abi.encode(uint256(keccak256("avalanche-ictt.storage.ERC20TokenHome")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 private constant _ERC20_TOKEN_HOME_STORAGE_LOCATION =
+        0x914a9547f6c3ddce1d5efbd9e687708f0d1d408ce129e8e1a88bce4f40e29500;
+
+    // solhint-disable ordering
+    function _getERC20TokenHomeStorage() private pure returns (ERC20TokenHomeStorage storage $) {
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            $.slot := _ERC20_TOKEN_HOME_STORAGE_LOCATION
```

Comment:
For the record, I think this usage of `$` is weird. It looks like you just followed the convention of the ERC-7201 example, but in my opinion it's a pretty bad name.

I'm averse to single-letter variables in general (though I've learned to accept them in Go), but at least when we use a letter the letter usually stands for something (eg `m` to stand for `message`). But `$` doesn't even stand for anything, so as a symbol it carries no inherent meaning/semantics.

And readers may not even know it's an allowable character in an identifier name in Solidity. I didn't. On my first read, I thought maybe it had some special meaning.

Because it carries no meaning, and is a potential source of confusion, I'd suggest avoiding `$` everywhere, opting instead for `_storage`.

Reply:
same as corresponding comment for teleporter PR, TLDR prefer to leave